test/rules/avoid_function_literal_in_foreach_method.dart

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N avoid_function_literal_in_foreach_method`
+
+void main() {
+  Iterable<String> people;
+
+  for (var person in people) { // OK
+    print(person);
+  }
+  people.forEach((person) { // LINT
```

Comment:
While I disagree with the style guide's recommendation (https://www.dartlang.org/guides/language/effective-dart/usage#avoid-using-iterableforeach-with-a-function-literal), the way I read it there is no distinction made on the basis of the form of the function literal. Hence, the lint, as written, conforms to the letter of the guideline (I can't speak to the spirit).

One could defeat the lint by converting the code into a `for` loop,
```
for (var p in people) {
  foo(p, 0);
}
```
by extracting the function into a local variable,
```
var f = (p) => foo(p, 0);
people.forEach(f);
```
or writing a local function
```
f(p) => foo(p, 0);
people.forEach(f);
```
but I agree that all three options seem counterproductive. (But that might be because, as I said, I don't agree with the guideline.)